### PR TITLE
Fix and finish test with the full workflow

### PR DESCRIPTION
**Pull request by szg251**
- Fix tests (explicit input routing, optional balancing)
- Factored out general txinfo builder
- Added a few more test cases, covering the full workflow (hopefully)

### CODE REVIEW

**Comment by szg251** [comment]
Sorry for the formatting changes, maybe my fourmolu is not picking up the config correctly...

File: coop-validation/test/Coop/Validation/Test/Transactions.hs
Comment (bladyjoker) [inline coop-validation/test/Coop/Validation/Test/Transactions.hs:1]: I would eventually put this in src/Coop/Transactions/<Certificates|Authentication|FactStatements.hs>. wdyt

File: coop-validation/test/Coop/Validation/Test/Transactions.hs [hunk 1]
```diff
@@ -409,14 +415,53 @@ addFactStatementActions mainTxInInfo mayCreateFss = do
...
+        ( \(fsTxInInfo, authTxInInfo, certTxInInfo) acts -> case factStatementFromUTxO ctx fsTxInInfo of
+            Nothing -> throwError $ CoopError'NotFactStatementInput fsTxInInfo
+            Just (fsId, fsDatum) -> do
+              let fsGcAfter = PAPI.factStatementDatum'gcAfter fsDatum
+              if strictLowerBound currentSlot > LowerBound fsGcAfter False
+                then setValidityAfter (Finite currentSlot)
+                else throwError $ CoopError'DeletingValidFactStatement fsId fsDatum currentSlot
+
+              certId <- case certificateFromUTxO ctx certTxInInfo of
+                Nothing -> throwError undefined
```
Comment (bladyjoker) [inline coop-validation/test/Coop/Validation/Test/Transactions.hs:432]: Throw `NotCertificateUTxO`

File: coop-validation/test/Coop/Validation/Test/Transactions.hs [hunk 2]
```diff
@@ -409,14 +415,53 @@ addFactStatementActions mainTxInInfo mayCreateFss = do
...
+                Just (certId, _certDatum) -> return certId
+
+              authValToBurn <- case find ((==) certId . fst) (authenticationFromUTxO ctx authTxInInfo) of
+                Just _ -> return $ authenticationValue ctx certId 1
+                _ -> throwError undefined
```
Comment (bladyjoker) [inline coop-validation/test/Coop/Validation/Test/Transactions.hs:437]: MissingAuthenticationValue

**Review by bladyjoker — APPROVED** [review]

File: coop-validation/test/Coop/Validation/Test/Transactions.hs
Comment (szg251) [reply coop-validation/test/Coop/Validation/Test/Transactions.hs:1]: Absolutely, I had a feeling that this is the plan

**Pull request merged**